src/js-api/handlers/ci-config.ts

```diff
@@ -70,9 +72,14 @@ export async function downloadCiConfig(args: DownloadCiConfigArgs) {
 
   const blob = await response.blob();
   const fileContents = await blob.text();
-  await fs.writeFile(metadata.ciConfigFilePath, fileContents, "utf-8");
+  const prefix = args.workingDir ?? "./";
+  const filePath = path.normalize(
+    path.resolve(prefix, metadata.ciConfigFilePath),
+  );
+  createDirs(path.resolve(filePath, "../"));
+  await fsP.writeFile(filePath, fileContents, "utf-8");
```

Comment:
Why is there a mixture of promise-based and synchronous file system operations between `createDirs` and `fsP.writeFile`?

Reply:
Since we added `create` command ci-config will be created in sub directories rather than CWD `createDirs` needs to finish (guaranteed finish) before writing ci-config or it failes to write ci-config (ci-config write was async and as it is the last file it can even gets created once the process finishes), but I can change one to have both async or sync if it works